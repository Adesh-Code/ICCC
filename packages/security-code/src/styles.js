const styles = ({ tokens, primitives }) => ({
  "&.DhigSecurityCode": {
    width: "56px",
    height: "72px",
    borderTop: `${primitives.borderWidths.medium} solid ${primitives.colors.tint["slate-450"]}`,
    borderBottom: "none"
  },
  '&.DhigSecurityCode [class*="MuiInputBase-input"][class*="MuiInputBase-inputAdornedEnd"]':
    {
      marginRight: "0"
    },
  "&.DhigSecurityCode[class*='MuiInput-root']:hover:before": {
    height: "2px",
    backgroundColor: tokens.colors["fill-engage"],
    boxShadow: "none",
    borderBottom: `${primitives.borderWidths.medium} solid ${primitives.colors.primary.black}`
  },
  "&.DhigSecurityCode:hover": {
    boxShadow: "none"
  },
  '&.DhigSecurityCode[class*="MuiInputBase-root"]': {
    boxShadow: `0px 1px 0px 0px ${primitives.colors.primary.white}`
  },
  '&.DhigSecurityCode[class*="MuiInputBase-root"]:before': {
    borderBottom: `1px solid ${primitives.colors.primary.black}`
  },
  "&.DhigSecurityCode input:focus": {
    boxShadow: `0px 2px 0px 0px ${tokens.colors["ink-focus"]} inset, 0px 0px 0px 2px ${tokens.colors["ink-focus"]} inset`
  },
  '&.DhigSecurityCode[class*="MuiInputBase-root"]:not(.Mui-disabled)::before': {
    boxShadow: "none"
  },
  '&.DhigSecurityCode [class*="MuiInputBase-input"]:before': {
    width: "56px"
  },
  "&.DhigSecurityCode input": {
    flexGrow: "1"
  },
  '&.DhigSecurityCode [class*="MuiInputBase-input"]': {
    padding: "0",
    width: "56px",
    height: "72px",
    fontSize: `${primitives.fontSizeFallbacks["6"]}`,
    fontWeight: primitives.fontWeights.default
  },
  '&.DhigSecurityCode--dense [class*="MuiInputBase-input"]': {
    padding: "0",
    width: "32px",
    height: "40px",
    fontSize: `${primitives.fontSizeFallbacks["3"]}`,
    fontWeight: primitives.fontWeights.default
  },
  "&.DhigSecurityCode--dense": {
    width: "32px",
    height: "40px"
  },
  '&.DhigSecurityCode--error[class*="MuiInputBase-root"]:before': {
    borderBottom: `1px solid ${primitives.colors.accent.clay}`
  },
  "&.DhigSecurityCode--error, &.DhigSecurityCode:first-of-type.DhigSecurityCode--error, &.DhigSecurityCode:last-child.DhigSecurityCode--error, &.DhigSecurityCode--error:before":
    {
      borderColor: primitives.colors.accent.clay
    },
  '&.DhigSecurityCode--error[class*="MuiInputBase-root"]:after': {
    boxShadow: `0px 1px 0px 0px ${primitives.colors.accent.clay}`
  },
  "&.DhigSecurityCode:first-of-type": {
    borderLeft: `${primitives.borderWidths.medium} solid ${primitives.colors.tint["slate-450"]}`
  },
  "&.DhigSecurityCode:last-child": {
    borderRight: `${primitives.borderWidths.medium} solid ${primitives.colors.tint["slate-450"]}`
  },
  "&.DhigSecurityCode:last-child [class*='MuiDivider-root'], &.DhigSecurityCode:focus-within [class*='MuiDivider-root']":
    {
      display: "none"
    }
});

export default styles;
