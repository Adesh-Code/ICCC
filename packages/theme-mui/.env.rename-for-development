# To load the MUI X Pro license for local development, rename this file to .env and replace with the actual license string
DHIG_MUIXPRO_ENTRY=PASTE_MUI_X_PRO_LICENSE_STRING_HERE