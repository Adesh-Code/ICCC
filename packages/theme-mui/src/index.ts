// Themes
export { default as themeDark } from "./themes/dark";
export { default as themeLight } from "./themes/light";

// Utilities
export { configureClassnameGenerator } from "./classname-generator";
export { setLicenseKey } from "./set-license-key";
