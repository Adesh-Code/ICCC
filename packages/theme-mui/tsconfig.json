{
  "extends": "../../tsconfig.json",
  "compilerOptions": {
    "outDir": "lib/esm",
    "resolveJsonModule": true,
    "declarationDir": "lib/types"
  },
  "include": ["src"]
}
