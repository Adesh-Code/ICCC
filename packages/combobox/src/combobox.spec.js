import React from "react";
import { fireEvent, render, screen } from "@testing-library/react";
import Combobox from "./combobox";

import { themeLight } from "@iccc/theme-mui";
import { createTheme, ThemeProvider } from "@mui/material/styles";

const light = createTheme(themeLight);

const onInputChangeMock = jest.fn();

const options = ["Alfa", "Bravo", "Charlie", "Delta", "Echo"];

const arrange = (props) => {
  const args = {
    id: "test-combobox",
    options,
    ...props
  };

  return render(
    <ThemeProvider theme={light}>
      <Combobox
        {...args}
        onInputChange={(event, value, reason) => onInputChangeMock(value)}
      />
    </ThemeProvider>
  );
};

describe("Combobox", () => {
  it("mounts the combobox component", () => {
    arrange();
    const combobox = screen.getByRole("combobox");
    expect(combobox).toMatchSnapshot();
  });

  it("doesn't automatically display dropdown on focus ", () => {
    arrange();
    const combobox = screen.getByRole("combobox");
    combobox.focus();
    expect(screen.queryByText("Alfa")).not.toBeInTheDocument();
  });

  it("displays dropdown when results of search query are found", async () => {
    arrange();
    const input = screen.getByRole("combobox");
    fireEvent.change(input, { target: { value: "a" } });
    expect(onInputChangeMock).toHaveBeenCalledWith("a");
    await screen.findByRole("listbox");
    expect(screen.getByText("Alfa")).toBeInTheDocument();

    // select a valid option from combobox dropdown
    fireEvent.click(screen.getByText("Alfa"));
    expect(onInputChangeMock).toHaveBeenCalledWith("Alfa");

    // open dropdown and check for selected option
    const button = screen.getByTitle("Open");
    fireEvent.click(button);
    await screen.findByRole("listbox");
    const selectedOption = (await screen.findAllByRole("option"))[0];
    expect(selectedOption.lastChild).toHaveClass("DhigCombobox--checkedIcon");
  });

  it("accepts classnames", () => {
    const { container } = arrange({
      className: "foo"
    });
    const combobox = container.firstChild;
    expect(combobox).toHaveClass("foo");
  });

  it("displays checkboxes in dropdown for multi-select", async () => {
    arrange({ multiple: true });
    const input = screen.getByRole("combobox");
    fireEvent.change(input, { target: { value: "a" } });
    const listbox = await screen.findByRole("listbox");
    const checkbox = await screen.getAllByRole("checkbox")[0];
    expect(listbox).toContainElement(checkbox);
  });

  it("displays correct fixed behaviour for multi-select", () => {
    const { container } = arrange({
      multiple: true,
      layout: "fixed",
      defaultValue: [options[0], options[1]]
    });
    const combobox = container.firstChild;
    expect(combobox).toHaveClass("DhigCombobox--multiple");
    expect(combobox).toHaveClass("DhigCombobox--layout--fixed");
  });

  it("displays correct grow behaviour for multi-select", () => {
    const { container } = arrange({
      multiple: true,
      layout: "grow",
      defaultValue: [options[0], options[1]]
    });
    const combobox = container.firstChild;
    expect(combobox).toHaveClass("DhigCombobox--multiple");
    expect(combobox).toHaveClass("DhigCombobox--layout--grow");
  });

  it("renders an addendum", async () => {
    arrange({ addendum: "Lorem ipsum" });
    const input = screen.getByRole("combobox");
    fireEvent.change(input, { target: { value: "a" } });
    await screen.findByRole("listbox");
    expect(screen.getByRole("presentation").firstChild.lastChild).toHaveClass(
      "DhigCombobox--addendum"
    );
  });

  it("displays correct error styles", () => {
    const { container } = arrange({
      error: true,
      defaultValue: options[0]
    });
    const combobox = container.firstChild.firstChild.firstChild;
    expect(combobox).toHaveClass("Mui-error");
  });
});
