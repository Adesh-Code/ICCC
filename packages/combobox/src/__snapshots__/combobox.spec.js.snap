// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Combobox mounts the combobox component 1`] = `
<input
  aria-autocomplete="list"
  aria-expanded="false"
  aria-invalid="false"
  autocapitalize="none"
  autocomplete="off"
  class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd DhigCombobox--input emotion-0"
  id="test-combobox"
  placeholder=""
  role="combobox"
  spellcheck="false"
  type="text"
  value=""
/>
`;
